identity-actor/src/actor/endpoint.rs

```diff
@@ -0,0 +1,141 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::borrow::Cow;
+use std::fmt::Display;
+use std::str::Split;
+
+use serde::Deserialize;
+use serde::Serialize;
+
+use crate::actor::Error;
+use crate::actor::Result as ActorResult;
+
+/// A path-like identifier for an actor request. As an example, `identity/create`
+/// could be an endpoint of the request "create" within the "identity" namespace.
+///
+/// The namespace and request are separated by a slash and only allow alphabetic ascii characters and `_`.
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+#[serde(try_from = "String", into = "String")]
+pub struct Endpoint {
+  name: Cow<'static, str>,
+}
+
+impl Endpoint {
+  /// Validates the endpoint.
+  fn validate(string: &str) -> ActorResult<()> {
+    let mut split: Split<'_, char> = string.split('/');
+
+    let namespace: &str = split.next().expect("split always returns at least one element");
```

Comment:
Could the use of `expect` here be an attack vector for malicious actors wanting to bring down a server?

Reply:
This comment still applies, but apparently it's still confusing. Not sure how to make it less confusing.

https://github.com/iotaledger/identity.rs/pull/322#discussion_r889674514

Reply:
Thanks for elaborating. We could consider adding a comment explaining that once the never type `!` lands the call to expect can hopefully be replaced with a match where the `None` variant is mapped to `!`.